lib/modules/manager/npm/post-update/yarn.ts

```diff
@@ -252,10 +252,40 @@ export async function generateLockFile(
           commands.push(`npx yarn-deduplicate --strategy ${s}`);
           // Run yarn again in case any changes are necessary
           commands.push(`yarn install${cmdOptions}`);
-        } else if (isYarnDedupeAvailable && s === 'highest') {
-          commands.push(`yarn dedupe --strategy ${s}${cmdOptions}`);
-        } else {
-          logger.debug(`yarn dedupe ${s} not available`);
+        } else if (s === 'highest') {
+          if (isYarnDedupeAvailable) {
+            commands.push(`yarn dedupe --strategy ${s}${cmdOptions}`);
+          } else {
+            const hasBothStrategies =
+              config.postUpdateOptions.includes('yarnDedupeFewer');
+            commands.push(
+              `yarn dlx yarn-berry-deduplicate${
+                hasBothStrategies ? '' : ` --strategy ${s}`
+              }`
+            );
+
+            // Run yarn again in case any changes are necessary
+            commands.push(`yarn install${cmdOptions}`);
+          }
+        } else if (s === 'fewer') {
+          const hasBothStrategies =
+            config.postUpdateOptions.includes('yarnDedupeHighest');
+
+          if (hasBothStrategies && !isYarnDedupeAvailable) {
+            // Skip this strategy as it will be run with "highest"
+            return;
+          }
+
+          commands.push(
+            `yarn dlx yarn-berry-deduplicate${
+              hasBothStrategies ? '' : ` --strategy ${s}`
+            }`
+          );
+
+          if (!hasBothStrategies || !isYarnDedupeAvailable) {
```

Comment:
are you sure about this condition? i think that needs to be removed and yarn install should always be called after dedupe.

Reply:
However I can remove if you're more comfortable without, it will mean more calls to yarn install even if not necessary